### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vladyslavchichkan**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,16 @@
...
+    def wrapper(*args) -> None:
+
+        if args in cache_dict:
+            print("Getting from cache")
+            return cache_dict[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[args] = result
+            return result
```
Comment (DevSpaciX) [inline app/main.py]: Think how you can call `return` only once . If cache_dict[args] = result you can return `cache_dict[args]` and it is makes 2 identical returns in your func . Optimise it : )

File: app/main.py [hunk 2]
```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args) -> None:
```
Comment (DevSpaciX) [inline app/main.py]: Add type hint for args , and change the None type ( You do not returning None )

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args) -> Any:
```
Comment (wQuelS) [inline app/main.py]: `*args` still miss type hint

File: app/main.py [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+
+        if args in cache_dict:
+            print("Getting from cache")
+            result = cache_dict[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[args] = result
+        return result
```
Comment (wQuelS) [inline app/main.py]: this also can be done without an additional `result` variable. This is not a crucial mistake and is totally **_up to you_**

**Review by wQuelS — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args) -> Callable:
+
```
Comment (wQuelS) [inline app/main.py:8]: a `Callable` object is, for example, a function. And that's not what your `wrapper` returns.
The previous `return type annotation` was correct, you had only to add a type hint for `*args`

**Review by wQuelS — CHANGES_REQUESTED** [review]

**Review by DevSpaciX — APPROVED** [review]
What okay ) But type hint for args is Any , and wrapper func returning Any type too : )

**Pull request closed without merging**